risc0/build/src/lib.rs

```diff
@@ -142,19 +141,17 @@ fn sha_digest_with_hex(data: &[u8]) -> (Vec<u8>, String) {
 }
 
 /// Returns the given cargo Package from the metadata.
-fn get_package<P>(manifest_dir: P) -> Package
-where
-    P: AsRef<Path>,
-{
+#[doc(hidden)]
```

Comment:
> 1. We should ban `#[doc(hidden)]` or have clear justification for its use.
> 2. Each PR should try to aim to address a single thing. This seems to have a mix of linker script changes along with `risczero build` support, which doesn't seem directly dependent on one another.

Ok, removed the "cargo risczero" changes from this PR

> 3. Without an explicit linker script, it's unclear where the DATA section is. There's an `_end` symbol that needs more documentation or a reference/link to how this symbol is defined. Do we know if this will work with all compilers?

I couldn't find any references to cite for it, but it works with LLVM, and as far as I know we don't have plans to support any other rust backends.

Newlib references it as following in a non-platform-specific header file, so it looks pretty standard to me:

`extern char _end[];                /* _end is set in the linker command file */